discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -437,10 +375,10 @@ impl FeedScope for SqliteStorage {
             .build()
             .persistent(false)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
-        Self::commit_tx(tx).await
+        tx.commit().await?;
+        Ok(())
```

Comment:
:art: this could be handled as a single expression
```suggestion
        tx.commit().await.map_err(Into::into)
```

Reply:
Through it doesn't convey the same meaning:

- "commit and handle the error"; "At the end of function return a empty ok result."
- "commit and return the result from the commit from the function while mapping the error"

The only reason why we can use `map_err` is because it happens that `commit` has the same return type as the function and it happens that the type only has one valid value.

Reply:
nonetheless this distinction not really needed here, imo. the commit can be regarded as the final operation of the method, hence the result of the commit is to a certain extend also the result of the method, hence it's not just error mapping, but also returning the result of the commit as the result of the method.

Reply:
<details>
<summary><- Me reasoning that this is not the case and we should generally use `?` where applicable.</summary>

it's not the same, the result of commit has nothing to do with the return value of `commit` as     long as it's not an error, e.g. in every case where we don't return `Ok(())` we return a value created before calling `commit()`.

More important look at if not from your or mine POV but of the POV of someone new to rust.

`?` is what they learn should always be used for error handling, `map_err` only appears much later in context of "when automatic error conversion doesn't work" , and even there extension traits like `context` and similar are often promoted over `.map_err`. For someone new to rust `.map_err(Into::into)` is only really nicely readable if they:

1. fully understand error handling 
2. understand the trait system
3. understand that methods are also free functions, and free functions are also closures

Or, more likely, learn it as another specific pattern.

Additionally it (minimally) increases the size of PRs changing the return type and in turn increases the chance for merge conflicts (negligible).

Sure both points are small, but I really can't agree that a stile which is harder to read for beginners and (negligible) bloats PRs without any benefit is better (or I'm missing the benefit).
</details>

I'm pretty sure we had that discussion before and for me it's both trivial to read so I will change it so that we can move on with this PR.